### PR TITLE
added code for new api with tests.

### PR DESCRIPTION
**Pull request by brysonthor**

### CODE REVIEW

File: birch-message.html [hunk 1]
```diff
@@ -212,8 +231,16 @@ <h4 header>[[i18n('new_message')]]</h4>
       _postMessage: function() {
         this.$.sendButton.loading = true;
         this.$.sendButton.disabled = true;
-        this._createBody();
-        this.$.sendReq.generateRequest();
+        console.log(this.fromUserId);
```
Comment (liechtym) [inline birch-message.html]: Should we delete this console.log?

File: birch-message.html [hunk 2]
```diff
@@ -212,8 +231,16 @@ <h4 header>[[i18n('new_message')]]</h4>
...
+        if(this.fromUserId){
+          this._createBody();
+          this.$.sendReq.url = this.ajaxBase + '/fst/fs-messages/threads';
+          this.$.sendReq.generateRequest();
+        } else {
+          this._createOldBody();
+          this.$.sendReq.url = this.ajaxBase + '/fst/mailbox/u2ms/api/v1/threads';
+          this.$.sendReq.generateRequest();
```
Comment (liechtym) [inline birch-message.html]: This could be outside of the if else statement since it's the same for both cases - code duplication

Comment (brysonthor) [inline birch-message.html]: good point, much thx

**Comment by brysonthor** [comment]
Ok I believe I've ironed out the tweaks, thanks for your help testing this with me.

**Comment by liechtym** [comment]
Thanks for the fixes @brysonthor. Merging.

**Pull request merged**